lib/membrane_mp4/muxer/cmaf.ex

```diff
@@ -3,18 +3,89 @@ defmodule Membrane.MP4.Muxer.CMAF do
+  segment will be 10 seconds long. Note that this doesn't apply to audio only tracks as with them you
+  can always create a segment withing a single sample bound.
+
+
+  > ### Important for video {: .warning}
+  >
+  > It is user's responsibility to properly set the segment duration range to align with
+  > whatever the keyframe interval of the video input is.
+
+  ## Partial segment creation
+  As previously mentioned, partial segments are not required to start with a key frame except for
+  a first partial of a new segment. Those are once again created based on the duration of the collected
+  samples but this time it has to be smarter as we can't allow the partial to significantly exceed
+  their target duration specified by `:partial_segment_duration_range`.
+
+  Exceeding the partial's target duration can cause unrecoverable player stalls e.g. when
+  playing LL-HLS on Safari, same goes if partial's duration is lower than 85% of target duration
+  when the partial is not last of its parent segment (Safari again). This is why it is important
+  that proper duration gets collected.
+
+  Besides respecting `:partial_segment_duration_range` when creating partials, we needs to take into
+  account the `:segment_duration_range` as eventually partials need to become part of a regular segment.
+
+  The behaviour of creating partials is as follows:
+  * if the duration of the **regular** segment currently being assembled is lower than the minimum then
+    try to collect **partial** with `target` value of `:partial_segment_duration_range` not matter what
+
+  * if the duration of the **regular** segment currently being assembled is greater than the minimum then try to
+    collect **partial** with at least a `minimum` value of `:partial_segment_duration_range`, up to the `target` value
+    but make sure that once a keyframe is encountered the partial immediately gets finalized
+
+  Note that once the `#{inspect(__MODULE__)}` is in a phase of finalizing a **regular** segment, the partial creation
+  needs to perform a lookahead of samples to make sure that we have the minimum duration and up to a target (we want to
+  avoid a situation where a keyframe will be placed before the minimum duration in a partial, therefore getting lost which will result
+  in several more partial getting created for the current segment).
+
+  Similarly to regular segments, if a key frame doesn't arrive for an extended period of time
+  then `#{inspect(__MODULE__)}` will keep on creating partial segments with a behaviour including the lookahead
+  (partial duration won't exceed the target but it may happen that couple of additional parts will get created).
+
+  > ### Important for video {: .warning}
+  >
+  > `:segment_duration_range` should be divisble by `:partial_segment_duration_range` for the best experience.
+  > For example when when a segment target duration is 6 seconds then partial target should be one of: 0.5s, 1s, 2s.
```

Comment:
This should be mentioned in the option's docs. Or we should have the target amount of partials per segment instead.